swap/tests/happy_path_restart_bob_before_comm.rs

```diff
@@ -1,16 +1,20 @@
 pub mod testutils;
 
-use swap::protocol::bob;
 use swap::protocol::bob::BobState;
+use swap::protocol::{alice, bob};
 use testutils::bob_run_until::is_xmr_locked;
 use testutils::SlowCancelConfig;
 
 #[tokio::test]
 async fn given_bob_restarts_after_xmr_is_locked_resume_swap() {
     testutils::setup_test(SlowCancelConfig, |mut ctx| async move {
-        let (bob_swap, bob_join_handle) = ctx.new_swap_as_bob().await;
+        let (bob_swap, bob_join_handle) = ctx.bob_swap().await;
+        let bob_swap = tokio::spawn(bob::run_until(bob_swap, is_xmr_locked));
 
-        let bob_state = bob::run_until(bob_swap, is_xmr_locked).await.unwrap();
+        let alice_swap = ctx.alice_swap_triggered_by_bob().await;
+        let alice_swap = tokio::spawn(alice::run(alice_swap));
+
+        let bob_state = bob_swap.await.unwrap().unwrap();
```

Comment:
Might be worth considering to allow this function to return errors, then you could make this less noisy by just doing `??`.